src/OneDrive.Sdk.Authentication.Common/Caching/CredentialCache.cs

```diff
@@ -20,8 +20,9 @@ public class CredentialCache
     {
         internal readonly IDictionary<CredentialCacheKey, AccountSession> cacheDictionary =
             new ConcurrentDictionary<CredentialCacheKey, AccountSession>();
+        internal CredentialCacheKey MostRecentlyUsedKey { get; set; }
 
-        private const int CacheVersion = 1;
+        private const int CacheVersion = 2;
```

Comment:
What do you mean? There's only one check. If cache is older than current version, discard it. Else, continue. That should never change. Also, given that this is sample code, so I don't think it's necessary to bend over backwards to save a couple caches (after all, it just requires going through the OAuth flow once more).